benches/bin/mvcctxn.rs

```diff
@@ -0,0 +1,220 @@
+extern crate rand;
+
+use tikv::storage::{new_local_engine, Engine, Key, Modify, Mutation, Options, SnapshotStore,
+                    Statistics, ALL_CFS, TEMP_DIR};
+use tikv::storage::mvcc::MvccTxn;
+use kvproto::kvrpcpb::{Context, IsolationLevel};
+
+use super::print_result;
+use test::BenchSamples;
+
+use rand::Rng;
+
+
+// TODO: Make this thread safe
+fn next_ts() -> u64 {
+    static mut CURRENT: u64 = 1;
+    unsafe {
+        CURRENT += 1;
+        CURRENT
+    }
+}
+
+#[inline]
+fn to_bytes(value: u64) -> Vec<u8> {
+    let mut result = Vec::new();
+    // big endian
+    for shift in (0..8).rev() {
+        result.push(((value >> shift) & 0xff) as u8);
+    }
+    result
+}
+
+fn create_row_key(tableid: u64, rowid: u64) -> Vec<u8> {
+    let mut result = to_bytes(tableid);
+    result.extend(b"_r".iter());
+    result.extend(to_bytes(rowid).iter());
+    // Key::from_raw(&result[..])
+    // result.append_ts(ts)
+    // result
+    result
+}
+
+fn generate_keys(count: usize) -> Vec<Vec<u8>> {
+    let mut result = Vec::new();
+    for i in 0..count {
+        result.push(create_row_key(1u64, i as u64));
+    }
+    result
+}
+
+#[inline]
+fn do_write(engine: &Engine, modifies: Vec<Modify>) {
+    engine.write(&Context::new(), modifies).unwrap();
+}
+
+#[inline]
+fn prewrite(engine: &Engine, mutations: &[Mutation], primary: &[u8], start_ts: u64) {
+    let snapshot = engine.snapshot(&Context::new()).unwrap();
+    let mut txn = MvccTxn::new(snapshot, start_ts, None, IsolationLevel::SI, false);
+    for m in mutations {
+        txn.prewrite(m.clone(), primary, &Options::default())
+            .unwrap();
+    }
+    do_write(engine, txn.into_modifies());
+}
+
+#[inline]
+fn commit(engine: &Engine, keys: &[Key], start_ts: u64, commit_ts: u64) {
+    let snapshot = engine.snapshot(&Context::new()).unwrap();
+    let mut txn = MvccTxn::new(snapshot, start_ts, None, IsolationLevel::SI, false);
+    for key in keys {
+        txn.commit(key, commit_ts).unwrap();
+    }
+    do_write(engine, txn.into_modifies());
+}
+
+fn prepare_test_engine(
+    versions: usize,
+    value_len: usize,
+    keys: &[Vec<u8>],
+    values: &mut Vec<Vec<u8>>,
+) -> Box<Engine> {
+    let engine = new_local_engine(TEMP_DIR, ALL_CFS).unwrap();
```

Comment:
I think we should the default TiKV configuration here.

Reply:
I saw the implementation  of  `new_local_engine` function, and I think it does used the default configurations.